src/lib/soup-to-svg.ts

```diff
@@ -40,56 +41,75 @@ function soupToSvg(soup: AnySoupElement[]): string {
       component.size,
       component.rotation || 0
     );
-    svgContent.push(svg);
+    svgChildren.push(svg);
     componentMap.set(component.schematic_component_id, component);
   }
 
   // Process ports and add lines to component edges
   for (const port of soup.filter((item) => item.type === "schematic_port")) {
     const flippedCenter = { x: port.center.x, y: flipY(port.center.y) };
     const svg = createSchematicPort(flippedCenter);
-    svgContent.push(svg);
+    svgChildren.push(svg);
 
     const component = componentMap.get(port.schematic_component_id);
     if (component) {
       const line = createPortToComponentLine(
         flippedCenter,
         component,
-        port.facing_direction
+        port.facing_direction || "right"
       );
-      svgContent.push(line);
+      svgChildren.push(line);
     }
   }
 
   // Process schematic traces
   for (const trace of soup.filter((item) => item.type === "schematic_trace")) {
     const svg = createSchematicTrace(trace, flipY, portPositions);
-    svgContent.push(svg);
+    if (svg) svgChildren.push(svg);
   }
 
   // Process text
   for (const text of soup.filter((item) => item.type === "schematic_text")) {
     const flippedPosition = { x: text.position.x, y: flipY(text.position.y) };
     const svg = createSchematicText(text, flippedPosition);
-    svgContent.push(svg);
+    svgChildren.push(svg);
   }
 
   const padding = 1;
   const width = maxX - minX + 2 * padding;
   const viewBox = `${minX - padding} ${minY - padding} ${width} ${height + 2 * padding}`;
 
-  return `
-    <svg xmlns="http://www.w3.org/2000/svg" viewBox="${viewBox}" width="1200" height="600">
-      <style>
-        .component { fill: none; stroke: red; stroke-width: 0.03; }
-        .component-pin { fill: none; stroke: red; stroke-width: 0.03; }
-        .trace { stroke: green; stroke-width: 0.03; fill: none; }
-        .text { font-family: Arial, sans-serif; font-size: 0.2px; }
-        .port { fill: none; stroke: blue; stroke-width: 0.03; }
-      </style>
-      ${svgContent.join("\n")}
-    </svg>
-  `;
+  const svgObject = {
+    name: 'svg',
+    type: 'element',
+    attributes: {
+      xmlns: 'http://www.w3.org/2000/svg',
+      viewBox,
+      width: '1200',
+      height: '600',
+    },
+    children: [
+      {
+        name: 'style',
+        type: 'element',
+        children: [
+          {
+            type: 'text',
+            value: `
+              .component { fill: none; stroke: red; stroke-width: 0.03; }
+              .component-pin { fill: none; stroke: red; stroke-width: 0.03; }
+              .trace { stroke: green; stroke-width: 0.03; fill: none; }
+              .text { font-family: Arial, sans-serif; font-size: 0.2px; }
+              .port { fill: none; stroke: blue; stroke-width: 0.03; }
```

Comment:
i think we should not use `<style />` and just put the properties directly on the svg objects, because theoretically this stuff will change for different layers, but issue for another day, just making a note